vector reserve before push_back

sduzh opened the pull request.

Commit:
vector reserve before push_back

1. reserve `SegmentWriter::_column_writers` before writing it
2. remove some condition branchs in SegmentWriter::init

imay commented on be/src/olap/rowset/segment_v2/segment_writer.cpp:
```diff
@@ -49,7 +49,9 @@ SegmentWriter::SegmentWriter(fs::WritableBlock* wblock,
 SegmentWriter::~SegmentWriter() = default;
 
 Status SegmentWriter::init(uint32_t write_mbytes_per_sec) {
+    (void)write_mbytes_per_sec;
```
```suggestion
Status SegmentWriter::init(uint32_t write_mbytes_per_sec __attribute__((unused))) {
```
also works

sduzh replied:
updated

imay commented:
@sduzh 
It will be better to describe what has been done in this CL in commit message.

imay reviewed (dismissed):
+1 LGTM

Commit:
fix hard-coded directory names in build-thirdpary.sh

replace `s2geometry-0.9.0` with `$S2_SOURCE`;
replace `CRoaring-0.2.60` with `CROARINGBITMAP_SOURCE`;
replace `orc-1.5.8` with `$ORC_SOURCE`
replace `cctz-2.3` with `$CCTZ_SOURCE`

imay commented:
Hi @sduzh 
Please don't push code to unrelated PR next time, better to create another PR.

kangkaisen reviewed (approved):
+1, LGTM

morningman reviewed (approved):
LGTM
P0 failed due to another bug, not related to this PR.
